### PR TITLE
Add Progress component

### LINKED ISSUE #924: Progress Component
## Overview

I need to create a component that has progress, `ProgressBar` today does not support what I need, so I needed to create a new component. Here is the prototype:

Screen Shot 2019-11-04 at 16 45 15

## Purpose

Here is the implementation of the new `Progress` component.

Screen Shot 2019-11-04 at 16 44 40

### PR DESCRIPTION
**Pull request by emersonlaurentino**
#### What is the purpose of this pull request?
#924

#### What problem is this solving?
closes #924

#### How should this be manually tested?

run styleguide locally

#### Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/Progress/styles.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.progressHeight {
+  height: 8px;
```
Comment (augustocb) [inline react/components/Progress/styles.css]: You should use `.5rem` instead!

Comment (emersonlaurentino) [reply react/components/Progress/styles.css]: nice, changed it. Thanks!

File: react/components/Progress/index.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import cn from 'classnames'
+import PropTypes from 'prop-types'
+import React from 'react'
+import styles from './styles.css'
+
+const propTypes = {
+  percent: PropTypes.number.isRequired,
+}
+
+const Progress: React.FC<PropTypes.InferProps<typeof propTypes>> = props => {
```
Comment (kaisermann) [inline react/components/Progress/index.tsx]: 👌

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+
+- `Progress` component
```
Comment (kevinch) [inline CHANGELOG.md]: Think we need to rename this component since we already got a `ProgressBar` component.

Comment (emersonlaurentino) [reply CHANGELOG.md]: yes, I will do this and maintain his compatibility with this one.

File: react/components/ProgressBar/index.tsx [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import Progress from '../Progress'
+import deprecated from '../../modules/deprecated'
+
+export default deprecated({
+  useNewComponent: {
+    old: 'ProgressBar',
+    new: 'Progress',
+  },
+})(Progress)
```
Comment (ohmyguigs) [inline react/components/ProgressBar/index.tsx]: yass! 🎉

**Pull request merged**